### DISCUSSION TITLE
How to rename file while uploading the file ?

### DISCUSSION THREAD

**Discussion by piotrh052b** [oneduo/nova-file-manager#122] [Q&A]
Hello !

Is it possible to rename file during upload e. g. for sanitize file name or when a file with that name already exists?

**Comment by Rezrazi** [comment]
Hi

No, applying modifications on the uploaded file on the fly is not supported
I would recommend hooking into either of these event and see if they fit your needs
- FileUploading
- FileUploaded


Docs: https://oneduo.github.io/nova-file-manager/events.html

Cheers

**Reply by 7ammer** [reply]
Trying to do this via `FileUploading` does not work. At least in the way I tried it:

```
    public function handle(FileUploading $event): void
    {
        // Make sure the path is url friendly
        $name = pathinfo($event->path, PATHINFO_FILENAME);
        $name = str($name)->slug();
        $extension = pathinfo($event->path, PATHINFO_EXTENSION);
        
        $event->path = $name . '.' . $extension;
    }
```

**Reply by 7ammer** [reply]
Same for `FiledUploaded`. No change:

```
    public function handle(FileUploaded $event): void
    {
        // Make sure the path is url friendly
        $name = pathinfo($event->path, PATHINFO_FILENAME);
        $name = str($name)->slug();
        $extension = pathinfo($event->path, PATHINFO_EXTENSION);
        $result = $name . '.' . $extension;
        
        $event->filesystem->move($event->path, $result);
    }
    ```

**Comment by DukeBJ** [comment]
Hi everyone!
I created my own Uploader handler by copying the class from `Oneduo\NovaFileManager\Filesystem\Upload\Uploader`. I added file renaming functionality to the `saveFile` method.

```
public function saveFile(UploadFileRequest $request, UploadedFile $file): array
    {
        ...
        // $filePath = $file->getClientOriginalName();
	$filePath = \Illuminate\Support\Str::slug(pathinfo($file->getClientOriginalName(),PATHINFO_FILENAME)).'.'.$file->getClientOriginalExtension();
	...
    }
```

And added my singleton to the `NovaServiceProvider`.

```
public function register(): void
    {
        parent::register();

	$this->app->singleton(\Oneduo\NovaFileManager\Contracts\Filesystem\Upload\Uploader::class, \App\NovaFileManager\Filesystem\Upload\Uploader::class);
    }
```

It's work for me.

<img width="467" height="547" alt="image" src="https://github.com/user-attachments/assets/10eadfe2-3fc7-44b6-be19-55c85b34c552" />
